octobot/community/supabase_backend/community_supabase_client.py

```diff
@@ -322,6 +324,91 @@ async def upsert_portfolio_history(self, portfolio_histories) -> list:
             on_conflict=f"{enums.PortfolioHistoryKeys.TIME.value},{enums.PortfolioHistoryKeys.PORTFOLIO_ID.value}"
         ).execute()).data
 
+    async def fetch_candles_history_range(
+        self, exchange: str, symbol: str, time_frame: commons_enums.TimeFrames
+    ) -> (typing.Union[float, None], typing.Union[float, None]):
+        min_max = json.loads(
+            (await self.postgres_functions().invoke(
+                "get_ohlcv_range",
+                {"body": {
+                    "exchange_internal_name": exchange,
+                    "symbol": symbol,
+                    "time_frame": time_frame.value,
+                }}
+            ))["data"]
+        )[0]
+        return (
+            self.get_parsed_time(min_max["min_value"]).timestamp() if min_max["min_value"] else None,
+            self.get_parsed_time(min_max["max_value"]).timestamp() if min_max["max_value"] else None,
+        )
+
+    async def fetch_candles_history(
+        self, exchange: str, symbol: str, time_frame: commons_enums.TimeFrames,
+        first_open_time: float, last_open_time: float
+    ) -> list:
+        total_candles_count = (last_open_time - first_open_time) // (
+            commons_enums.TimeFramesMinutes[time_frame] * commons_constants.MINUTE_TO_SECONDS
+        )
+        offset = 0
+        max_size = 0
+        total_candles = []
+        max_requests_count = 100
+        request_count = 0
+        while request_count < max_requests_count:
+            request = (
+                self.table("temp_ohlcv_history").select("*")
+                .eq(
+                    "exchange_internal_name", exchange
+                ).eq(
+                    "symbol", symbol
+                ).eq(
+                    "time_frame", time_frame.value
```

Comment:
You can use `match` instead of multiple `eq`

Reply:
up, thanks !